### ISSUE TITLE
Package fails to install under R 4.0

### ISSUE THREAD

**Issue by klmr** [pharmaverse/admiraldev#456]
According to its `DESCRIPTION`, the package requires R 4.0. However, the following line uses the native pipe operator, which is only available from R 4.1 onwards:

https://github.com/pharmaverse/admiraldev/blob/86c007a71a99a417ac9b8dbbc36672afc0e8a1b0/R/process_set_values_to.R#L96

**Comment by bms63** [comment]
Ohhh...goodspot @klmr !!

@pharmaverse/admiral and @pharmaverse/admiral_comm 

 ummm....how should we approach:

* Remove native pipe and update to magrittr pipe
* Require admiral/admiraldev to only work with 4.1  and keep native pipe?

**Comment by manciniedoardo** [comment]
> Ohhh...goodspot @klmr !!
> 
> @pharmaverse/admiral and @pharmaverse/admiral_comm
> 
> ummm....how should we approach:
> 
> * Remove native pipe and update to magrittr pipe
> * Require admiral/admiraldev to only work with 4.1  and keep native pipe?

I'd vote for the second option - R 4.1 is quite old anyway.

**Comment by bms63** [comment]
So should we handle this as an official bug fix with a re-release?

**Comment by ddsjoberg** [comment]
> So should we handle this as an official bug fix with a re-release?

I don't think so. The min R version is something that is not correct is many many many pkgs on CRAN. I would just fix in dev, and it'll be correct in the next release

**Comment by bundfussr** [comment]
Could we add a CI/CD workflow which tries to install and test the package with the minimum versions stated in `DESCRIPTION`?

**Comment by ddsjoberg** [comment]
Perhaps we can just say somewhere on the website we support the last 2, 3, 4 (whatever we choose) versions of R, and supplement our workflows with the R CMD Check from Posit that makes it easy to check the last n versions of R.

```yaml
          - {os: ubuntu-latest,   r: 'devel', http-user-agent: 'release'}
          - {os: ubuntu-latest,   r: 'release'}
          - {os: ubuntu-latest,   r: 'oldrel-1'}
          - {os: ubuntu-latest,   r: 'oldrel-2'}
          - {os: ubuntu-latest,   r: 'oldrel-3'}
          - {os: ubuntu-latest,   r: 'oldrel-4'}
```

**Comment by millerg23** [comment]
We say we develop using the earliest of the last 3 versions of R, in documentation:
https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#r-and-package-versions-for-development

**Comment by bms63** [comment]
I believe we are testing 4.4, 4.3, 4.2 with our CI actions.  We should add 4.1 to mix.  

![image](https://github.com/pharmaverse/admiraldev/assets/10111024/fa5387d0-9297-46c7-875b-98e3e633d81f)

**Comment by ddsjoberg** [comment]
We already have the Posit workflow file here, and we can just uncomment the lines.
https://github.com/pharmaverse/admiraldev/blob/86c007a71a99a417ac9b8dbbc36672afc0e8a1b0/.github/workflows/R-CMD-check.yaml#L32

**Comment by bms63** [comment]
ah i see - ummm... can we re-work the admiralci one since it is used in admiral as well as well as the other admiral packages.  Just be nice to all pull from the same spot.  I'm totally fine to simplify the one in admiralci!

**Comment by bundfussr** [comment]
> We say we develop using the earliest of the last 3 versions of R, in documentation: https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#r-and-package-versions-for-development

There it says
> Currently we test [admiral](https://pharmaverse.github.io/admiral/) against the three latest R Versions and the closest snapshots of packages to those R versions.

Is the part "and the closest snapshots of packages to those R versions" up to date? I remember vaguely that we changed our strategy to use the latest version of the packages for all R versions.

**Comment by bundfussr** [comment]
> I believe we are testing 4.4, 4.3, 4.2 with our CI actions. We should add 4.1 to mix.

I think testing more R versions doesn't solve the issue. If we had tested with 4.1, we still wouldn't have discovered the problem.

If we can add a workflow which tests the package on the minimum R version and minimum package versions as specified in the `DESCRIPTION` file, it would be helpful. @cicdguy , is this possible?

Otherwise, I would rely on the developers and users that they report wrong minimum versions.

**Comment by bms63** [comment]
> > I believe we are testing 4.4, 4.3, 4.2 with our CI actions. We should add 4.1 to mix.
> 
> I think testing more R versions doesn't solve the issue. If we had tested with 4.1, we still wouldn't have discovered the problem.
> 
> If we can add a workflow which tests the package on the minimum R version and minimum package versions as specified in the `DESCRIPTION` file, it would be helpful. @cicdguy , is this possible?
> 
> Otherwise, I would rely on the developers and users that they report wrong minimum versions.

My understanding is that we are saying in our current DESCRIPTION that admiral works with R version 4.0.  However, we are using a pipe `|>` that was introduced in R version 4.1 in a few lines of code

If we had been testing the package build on 4.0 I believe we would of found this issue out.  Maybe we should just give a try to experiment.

Anyways, I do think the minimum version we suggest in the DESCRIPTION should be tested no matter what! :)
